src/cargo/core/manifest.rs

```diff
@@ -28,13 +29,15 @@ pub struct Manifest {
     profiles: Profiles,
     publish: bool,
     replace: Vec<(PackageIdSpec, Dependency)>,
+    augment: HashMap<Url, Vec<Dependency>>,
```

Comment:
Yeah the intention here was that the syntax of this section is the same as `[dependencies]`, so we'd just store it as a vanilla `Dependency`. Then it only makes sense to augment with git sources or path sources, which only ever return one summary anyway, so it'd suffice for that.